### ISSUE TITLE
Subclass of Tuple[X, ...] rejects Iterable[X] in argument to __new__

### ISSUE THREAD

**Issue by Syeberman** [python/mypy#8957]
I believe I'm getting an incorrect error from mypy 0.780. The following code:

```python
from typing import Tuple, Iterable

class NineCell:
    pass

class NineHouse(Tuple[NineCell, ...]):
    name: str

    def __new__(cls, name: str, iterable: Iterable[NineCell]) -> "NineHouse":
        self = super().__new__(cls, iterable)
        self.name = name
        return self
```

produces this error:

```
error: Argument 2 to "__new__" of "tuple" has incompatible type "Iterable[NineCell]"; expected "Iterable[_T_co]"
```

**Comment by ilevkivskyi** [comment]
Yeah, looks like a bug, but probably low priority.

**Comment by kveretennicov** [comment]
Still present as of 0.991

Here is the use case and error example: https://gist.github.com/mypy-play/1e7817d3b383b6f59f65b643a5e78d44

```
from __future__ import annotations

from collections.abc import Set

from typing import TypeVar
_T_co = TypeVar("_T_co", covariant=True)


class UniqueSequence(tuple[_T_co]):
    def __new__(cls, source: Set[_T_co] | UniqueSequence[_T_co]) -> UniqueSequence[_T_co]:
        return super().__new__(cls, source)
```

> Argument 2 to "__new__" of "tuple" has incompatible type "Union[AbstractSet[_T_co], UniqueSequence[_T_co]]"; expected "Iterable[_T_co]"  [arg-type]

**Comment by genovese** [comment]
I am still encountering this problem in mypy 1.4.1 and Python 3.10.

```
T = TypeVar('T', bound=Numeric)

class VecTuple(tuple[T,...]):
    def __new__(cls, contents: Iterable[T]) -> VecTuple[T]:
        return super().__new__(cls, contents)
```
Two messages: `Argument 2 to "__new__" of "tuple" has incompatible type "Iterable[T]"; expected "Iterable[_T_co]"  [arg-type]` and `Value of type variable "Self" of "__new__" of "tuple" cannot be "VecTuple[T]"  [type-var]`.

Neither changing the variance nor eliminating the bound, alone or in combination, eliminate the problem.  Thanks

**Comment by kveretennicov** [comment]
All failure examples given above now pass, starting with mypy 1.9.0 :raised_hands: This should be closed.
